### PR TITLE
Create ConfigMap for cluster metadata and add ConfigMap details to xpk inspector

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
## Fixes / Features
- Create a cluster ConfigMap to store metadata like `xpk_version`(PyPI version of xpk)
- Add cluster ConfigMap details to xpk inspector output

## Testing / Documentation
Testing details.

- [ y ] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: xpk.py [hunk 1]
```diff
@@ -1148,32 +1151,51 @@ def create_cluster_configmap(args, system):
...
     0 if successful and 1 otherwise.
   """
+  configmap_yml = {}
   device_type = args.tpu_type if args.tpu_type else args.device_type
-  data = f'{device_type}: "{int(args.num_slices) * system.vms_per_slice}"'
-  yml_string = cluster_configmap_yaml.format(args=args,
-                                           data=data)
-  tmp = write_temporary_file(yml_string)
-  command = f'kubectl apply -f {str(tmp.file.name)}'
+  resources_data = f'{device_type}: "{int(args.num_slices) * system.vms_per_slice}"'
```
Comment (Obliviour) [inline xpk.py:1156]: nit to add two comments to explain that there are two config maps being created. One on ln 1156 and another on ln 1163.

File: xpk.py [hunk 2]
```diff
@@ -1148,32 +1151,51 @@ def create_cluster_configmap(args, system):
...
+  task_names = []
+  for configmap_name, yml_string in configmap_yml.items():
+    tmp = write_temporary_file(yml_string)
+    command = f'kubectl apply -f {str(tmp.file.name)}'
+    commands.append(command)
+    task_name = f'ConfigMapCreate-{configmap_name}'
+    task_names.append(task_name)
 
-  return_code = run_command_with_updates(command, 'GKE Cluster Create ConfigMap', args)
+  return_code = run_commands(commands, 'GKE Cluster Create ConfigMap', task_names)
```
Comment (Obliviour) [inline xpk.py:1179]: nit GKE Cluster Create ConfigMap(s)

File: xpk.py [hunk 3]
```diff
@@ -1148,32 +1151,51 @@ def create_cluster_configmap(args, system):
   Returns:
```
Comment (Obliviour) [inline xpk.py:1151]: I think it might be worth renaming this function to `create_cluster_configmaps` ? since we are now creating more than one?

**Review by Obliviour — APPROVED** [review]
LGTM, just some small changes. 

You should also see a new build test that runs with incoming changes. LMK if you have unexpected failures in that.

File: xpk.py (hunk 3 above)
Comment (SurbhiJainUSC) [reply xpk.py:1151]: Done!

File: xpk.py (hunk 1 above)
Comment (SurbhiJainUSC) [reply xpk.py:1158]: Done!

**Pull request merged**